elfo-logger/src/line_buffer.rs

```diff
@@ -0,0 +1,274 @@
+use std::mem;
+
+use crate::line_transaction::Line;
+
+// BufferConfiguration
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub(crate) struct BufferConfig {
+    pub(crate) max_line_size: usize,
+}
+
+impl Default for BufferConfig {
+    fn default() -> Self {
+        Self {
+            max_line_size: usize::MAX,
+        }
+    }
+}
+
+// Repr
+
+#[derive(Debug)]
+struct Repr<'a> {
+    buf: &'a mut LineBuffer,
+
+    // We don't wanna write unfinished data, so there must be a way to
+    // revert changes made by `CurrentLine`
+    pre_start_buffer_size: usize,
+}
+
+// CurrentLine
+
+#[derive(Debug)]
+pub(crate) struct CurrentLine<'a>(Repr<'a>);
```

Comment:
Rename to something more meaningful?

Reply:
Done in [c2495f0](https://github.com/elfo-rs/elfo/pull/128/commits/c2495f0651fe992875ff299919774d87a0905418)